apis/cloudscale/v1/bucket_types.go

```diff
@@ -44,6 +54,15 @@ type BucketParameters struct {
 	// The region must be available in the S3 endpoint.
 	// Cannot be changed after bucket is created.
 	Region string `json:"region"`
+
+	// +kubebuilder:validation:Enum=DeleteIfEmpty;DeleteAll
+	// +kubebuilder:default="DeleteIfEmpty"
+
+	// BucketDeletionPolicy determines how buckets should be deleted when Bucket is deleted.
+	//  `DeleteIfEmpty` only deletes the bucket if the bucket is empty.
+	//  `DeleteAll` recursively deletes all objects in the bucket and then removes it.
+	// To skip deletion of the bucket (orphan it) set `spec.deletionPolicy=Orphan`.
```

Comment:
This is really weird from a user perspective. Having two different params that configure different aspect of the same feature? It will also make it more difficult to handle in a composition later on.

We should probably allow `spec.forProvider.deletionPolicy=Orphan` and then populate `spec.deletionPolicy` accordingly.

E.g. 
`DeleteIfEmpty` or `DeleteAll ` -> `spec.deletionPolicy=Delete`
`Orphan` -> `spec.deletionPolicy=Orphan`

Reply:
We can't extend the possible values of `spec.deletionPolicy`, they're defined in the crossplane-runtime.
If `spec.deletionPolicy` is `Orphan` then `spec.forProvider.bucketDeletionPolicy` has no meaning anyway (it's like removing the finalizer).
It may seem weird, but what's even weirder is when we update `spec.deletionPolicy`  from `spec.forProvider.bucketDeletionPolicy`. Also, if both policies have `Orphan`, which one is now authoritative? As soon as `spec.deletionPolicy` is `Orphan` and the resource is deleted, Crossplane doesn't even bother anymore about the `forProvider` spec.

Reply:
I don't think the Crossplane Authors have a design in mind where the managed resource has additional deletion policies, or a security feature that needs confirmation for deleting it (`DeleteIfEmpty` prevents data deletion).
If you think this additional policy is weird, then we should consider _always_ to delete the bucket, regardless of the data in it (I presume that's Crossplane by design). It means we remove `spec.forProvider.bucketDeletionPolicy` completely. But this would potentially make gradual deletions (e.g. keep backup data for a period of time) harder.

Reply:
I see. In that case we can keep it as is and treat `Orphan` as a special case and not expose it as an option in the XRD.